etcdserver/etcdhttp/client_security.go

```diff
@@ -51,22 +51,25 @@ func hasRootAccess(sec *security.Store, r *http.Request) bool {
 	if !ok {
 		return false
 	}
-	if username != "root" {
-		log.Printf("security: Attempting to use user %s for resource that requires root.", username)
-		return false
-	}
-	root, err := sec.GetUser("root")
+	rootUser, err := sec.GetUser(username)
 	if err != nil {
 		return false
 	}
-	ok = root.CheckPassword(password)
+	ok = rootUser.CheckPassword(password)
 	if !ok {
 		log.Printf("security: Wrong password for user %s", username)
+		return false
+	}
+	for _, role := range rootUser.Roles {
+		if role == security.RootRoleName {
+			return true
+		}
 	}
-	return ok
+	log.Printf("security: Attempting to use user %s for resource that requires %s role.", username, security.RootRoleName)
```

Comment:
Perhaps: "security: User %s does not have the %s role which is required to access %s resource"?